core/src/main/java/org/apache/iceberg/avro/ValueWriters.java

```diff
@@ -373,6 +382,82 @@ public void write(BigDecimal decimal, Encoder encoder) throws IOException {
     }
   }
 
+  private abstract static class VariantBinaryWriter<T> implements ValueWriter<T> {
+    private ByteBuffer reusedBuffer = ByteBuffer.allocate(2048).order(ByteOrder.LITTLE_ENDIAN);
+
+    protected abstract int sizeInBytes(T value);
+
+    protected abstract int writeTo(ByteBuffer buffer, int offset, T value);
+
+    @Override
+    public void write(T datum, Encoder encoder) throws IOException {
+      if (datum instanceof Serialized) {
+        encoder.writeBytes(((Serialized) datum).buffer());
```

Comment:
I assume this is why we're making `Serialized` public? I was looking to see if there were some other way, but this looks as good as anything.

Reply:
Yeah, that's correct. It seems reasonable to me to avoid the copy.